gctree/branching_processes.py

```diff
@@ -62,7 +72,7 @@ def __init__(self, tree: ete3.TreeNode = None, allow_repeats: bool = False):
             observed_genotypes.add(self.tree.name)
```

Comment:
This whole section can be drastically simplified I think, since we no longer need to accommodate bootstrap samples, and so maybe don't need to accumulate node names up the tree as sets. Is there any issue with removing this and just retaining one node name on edge collapse, no matter what?

Root-unifurcating trees added to the history DAG get an unobserved dummy leaf added below root (the history DAG doesn't allow unifurcation), so the logic in this section results in a CollapsedTree with a strange double-labeled root node after edge collapse. This can be fixed elsewhere (_clade_tree_to_ctree) if we don't want to change this section.

Reply:
This is already resolved in `_clade_tree_to_ctree` so we can just ignore this whole idea for now if we want.